packages/account/src/BiconomySmartAccountV2.ts

```diff
@@ -165,6 +169,32 @@ export class BiconomySmartAccountV2 extends BaseSmartAccount {
    * calculate the account address even before it is deployed
    */
   async getCounterFactualAddress(params?: CounterFactualAddressParam): Promise<string> {
+    const validationModule = params?.validationModule ?? this.defaultValidationModule;
+    const index = params?.index ?? this.index;
+
+    // Note: Review
+    // the fact that below flow (AddressResolver) is Always called can be avoided by passing a flag in smart account config
+    // (if it's intended to detect V1 upgraded accounts)
+
+    // is instanceOf ECDSAOwnershipValidationModule or address matches ECDSA module address
+    if (validationModule.getAddress() === DEFAULT_ECDSA_OWNERSHIP_MODULE) {
+      const eoaSigner = await validationModule.getSigner();
+      const eoaAddress = await eoaSigner.getAddress();
+      const accountAddress = await this.getV1AccountsUpgradedToV2(eoaAddress, index);
+      Logger.log("account address from V1 ", accountAddress);
+      if (accountAddress !== ethers.constants.AddressZero) {
+        return accountAddress;
+      }
+    } else {
+      // can check using module.getAddress() and module.getInitData() for some other auth module
+    }
```

Comment:
Why not keep it generic, take signer and call getAddress with `module.getAddress()` and `module.getInitData()`

Reply:
hmm ok in that case I can always call then this function instead.

https://github.com/bcnmy/scw-contracts/blob/features/SMA-321-account-address-resolver/contracts/smart-account/utils/AddressResolver.sol#L95

Reply:
I am also thinking about more specific methods in the contract and have another review.

like I added this comment
https://github.com/bcnmy/scw-contracts/blob/features/SMA-321-account-address-resolver/contracts/smart-account/utils/AddressResolver.sol#L19C5-L19C76 
What do you think @AmanRaj1608 @ankurdubey521 

because here we're and will always be only interested in querying possible addresses from V1 factory which have implementation as V2. so why query V2 addresses?

Reply:
I feel both is fine, if removing the query of v2 address can save some time let's remove it.